skill_tagging/pipeline.py

```diff
@@ -0,0 +1,79 @@
+"""
+Module that contains the openedx_filters pipeline steps.
+"""
+import logging
+import random
+
+import pkg_resources
+from django.conf import settings
+from django.template import Context, Template
+from openedx_filters import PipelineStep
+
+logger = logging.getLogger(__name__)
+DEFAULT_PROBABILITY = 0.5
+
+
+class AddVerticalBlockSkillVerificationSection(PipelineStep):
+    """
+    Adds extra HTML to the fragment.
+
+    Example Usage:
+
+    .. code-block::
+
+        "OPENEDX_FILTERS_CONFIG": {
+            "org.openedx.learning.vertical_block.render.completed.v1": {
+                "fail_sliently": false,
+                "pipeline": [
+                    "skill_tagging.pipeline.AddVerticalBlockSkillVerificationSection"
+                ]
+            }
+        }
+    """
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def fetch_related_skills(self, block):
+        """Checks `has_verified_tags` and fetchs related skills."""
+        has_verified_tags = getattr(block, "has_verified_tags", None)
+        if has_verified_tags is None or has_verified_tags is True:
+            return []
+        fetch_tags = getattr(block, "fetch_skill_tags", None)
+        if fetch_tags is None:
+            return []
+        tags = fetch_tags()
+        return tags
+
+    def should_run_filter(self, skills):
+        """Determines whether we should run filter and display form."""
+        # return false if no skills found.
+        if not skills:
```

Comment:
If the skills is only passed here to check if it's empty or not, it might be better to have the check before calling this function (see suggestion below)